ui/app/mainui/AppMain.qml

```diff
@@ -218,14 +218,111 @@ Item {
+                        icon_ = "warning"
+                    }
+                    Global.displayToastMessage(toastTitle_, toastSubtitle_, icon_, !error, toastType_, urlLink_)
+                }
+                break
+            default:
+                if (!error) {
+                    Global.displayToastMessage(qsTr("Transaction pending..."),
+                                               qsTr("View on etherscan"),
+                                               "",
+                                               true,
+                                               Constants.ephemeralNotificationType.normal,
+                                               "%1/%2".arg(appMain.rootStore.getEtherscanLink(chainId)).arg(txHash))
+                }
+                break
+            }
+        }
+
+        function onShowToastTransactionSendingComplete(chainId: int, txHash: string, data: string, success: bool,
+                                                       txType: int, fromAddr: string, toAddr: string, fromTokenKey: string,
+                                                       fromAmount: string, toTokenKey: string, toAmount: string) {
+            switch(txType) {
+            case Constants.SendType.Approve:
+                let fromToken = ModelUtils.getByKey(appMain.tokensStore.plainTokensBySymbolModel, "key", fromTokenKey)
+                let fromAccountName = ModelUtils.getByKey(appMain.transactionStore.accounts, "address", fromAddr, "name")
+                let networkName = ModelUtils.getByKey(WalletStore.RootStore.filteredFlatModel, "chainId", chainId, "chainName")
+                if(!!fromToken && !!fromAccountName && !!networkName) {
+                    let approvalAmount = currencyStore.formatCurrencyAmountFromBigInt(fromAmount, fromToken.symbol, fromToken.decimals)
+                    let toastTitle = qsTr("Spending cap set: %1 in %2 for %3 on %4").arg(approvalAmount).arg(fromAccountName).arg(Constants.swap.paraswapUrl).arg(networkName)
+                    let toastSubtitle =  qsTr("View on %1").arg(networkName)
+                    let urlLink = "%1/%2".arg(appMain.rootStore.getEtherscanLink(chainId)).arg(txHash)
+                    let toastType = Constants.ephemeralNotificationType.success
+                    let icon = success  ? "checkmark-circle" : "warning"
+                    if(!success) {
+                        toastTitle = qsTr("Failed to set spending cap: %1 in %2 for %3 on %4").arg(approvalAmount).arg(fromAccountName).arg(Constants.swap.paraswapUrl).arg(networkName)
+                        toastType = Constants.ephemeralNotificationType.danger
+                    }
+                    Global.displayToastMessage(toastTitle, toastSubtitle, icon, false, toastType, urlLink)
+                }
+                break
+            case Constants.SendType.Swap:
+                let fromToken_ = ModelUtils.getByKey(appMain.tokensStore.plainTokensBySymbolModel, "key", fromTokenKey)
```

Comment:
Wondering why the underscore at the end... do you get some conflict/warning with the case above? That shouldn't happen tho, it's a different context/scope  :thinking:

Reply:
yes I was, but I didnt add `{}` brackets, now I have so not needed anymore